### ISSUE TITLE
Adding periodic boundary conditions to watershed

### ISSUE THREAD

**Issue by kai-luca** [scikit-image/scikit-image#7046]
### Description:

Adding periodic boundary conditions for the watershed transform would be an easy, yet practicable thing to do.
It would make this library more usable for people using it to sort their simulation data(with periodic boundary conditions) into clusters.
It also would be easy to implement since at the moment the picture is explicitly padded to remove periodic boundary conditions.
Just make this padding the default but removable via function parameter.

**Comment by stefsmeets** [comment]
Just chiming in that this would also be useful for one of the problems I'm working on.

I tested the suggested approach (remove the padding), and it nicely wraps around the array.

See test code: https://gist.github.com/stefsmeets/c7bf4f5c2fcee9ec9d7ae688e782ce9f

**Comment by stefsmeets** [comment]
No worries. From my view, this would be useful for a host of image analysis problems in the material sciences. For example when working with volumes/densities from simulations with periodic boundary conditions.

**Comment by jni** [comment]
Yeah, indeed the wrap-around without the padding is *not* the wrap-around you want for periodic boundary conditions. For example, if you have an array of shape (10, 10) and you are at position (5, 0), your linearised neighbours are (4, 0), (6, 0), (5, 1) (as expected) and (4, 9) — here the 4 is unexpected. So you still need to do some arithmetic to do the wrapping correctly. However, I would definitely support a PR to add this feature, I think it's an excellent idea.
